org/mozilla/jss/pkcs11/PK11ECPublicKey.java

```diff
@@ -4,49 +4,37 @@
 
 package org.mozilla.jss.pkcs11;
 
-// Requires JAVA 1.5
-//import java.security.interfaces.ECPublicKey;
+import java.security.interfaces.ECPublicKey;
+import java.security.spec.ECPoint;
+import java.security.spec.ECParameterSpec;
 import java.math.BigInteger;
 
-//
-// Requires JAVA 1.5
-//public final class PK11ECPublicKey extends PK11PubKey implements ECPublicKey {
-public final class PK11ECPublicKey extends PK11PubKey {
+import org.mozilla.jss.netscape.security.util.Utils;
+import org.mozilla.jss.util.EC;
+
+public final class PK11ECPublicKey extends PK11PubKey implements ECPublicKey {
 
     private static final long serialVersionUID = 1L;
     public PK11ECPublicKey(byte[] pointer) {
         super(pointer);
     }
 
-//
-// Requires JAVA 1.5
-//    public ECParams getCurve() {
-//      try {
-//        return new BigInteger( getCurveByteArray() );
-//      } catch(NumberFormatException e) {
-//          throw new RuntimeException("Unable to decode DSA parameters: " + e.getMessage(), e);
-//      }
-//    }
-//
-
-    public BigInteger getCurve() {
-      try {
-        return new BigInteger( getCurveByteArray() );
-      } catch(NumberFormatException e) {
-          throw new RuntimeException("Unable to decode EC curve: " + e.getMessage(), e);
-      }
+    public ECParameterSpec getParams() {
+        byte[] curveData = getCurveByteArray();
+        return EC.decodeNSSOID(curveData);
     }
 
     public byte[] getCurveBA() {
         return getCurveByteArray();
     }
 
-    public BigInteger getW() {
-      try {
-        return new BigInteger( getWByteArray() );
-      } catch(NumberFormatException e) {
-          throw new RuntimeException("Unable to decode EC public value: " + e.getMessage(), e);
-      }
+    public ECPoint getW() {
+        byte[] pointData = getWByteArray();
+        return EC.decodeNSSPoint(pointData);
+    }
+
+    public byte[] getWBA() {
```

Comment:
Instead of adding getWBA() we could simply make getWByteArray() public.

Reply:
Ah, sure, good catch.

Reply:
I'm confused why this is actually needed. I might remove it in the future, but for now I've done this.